packages/salesforce-adapter/src/filters/custom_objects_instances.ts

```diff
@@ -591,6 +606,20 @@ const filterCreator: RemoteFilterCreator = ({ client, config }) => ({
       customObjectFetchSetting,
       setting => setting.invalidIdFields.length === 0 && setting.invalidManagedBySaltoField === undefined
     )
+
+    if (validFetchSettings.length === 0
+    && invalidFetchSettings.length > 0
+    && invalidFetchSettings.every(setting => setting.invalidManagedBySaltoField !== undefined)) {
+      return {
+        errors: [
+          {
+            message: 'The Salto environment is configured to use a Salto Management field but the field is missing or is of the wrong type for all data records. No data records will be fetched. Please verify the field name in saltoManagementFieldSettings.defaultFieldName in the environment configuration file.',
```

Comment:
```suggestion
            message: 'The Salto environment is configured to use the Salto Management field "${fieldName}" which seem to be invalid on all of the Custom Objects. No data records will be fetched. Please verify this field exists and is of supported type.',
```
I think we can improve the message here. Also, @pgonzaleznetwork @tomermevorach will we have an article explaining this? Perhaps we can include it later as part of this warning with more detailed explanation.

Reply:
@tomermevorach ? @pgonzaleznetwork ?